numpy/core/_dtype_ctypes.py

```diff
@@ -61,12 +60,24 @@ def _from_ctypes_structure(t):
     else:
         fields = []
         for fname, ftyp in t._fields_:
-            fields.append((fname, dtype_from_ctypes_type(ftyp)))
+            fields.append((fname, dtype_from_ctypes_endianness(ftyp)))
```

Comment:
There's no need to change this line or the one above - the recursion will handle it just fine

Reply:
Indeed. Changed.